src/lib.rs

```diff
@@ -2240,6 +2276,62 @@ mod tests {
         }
     }
 
+    #[cfg(feature = "ndarray")]
+    #[test]
+    fn test_tensor_from_ndarray() {
+        let arr = Array::<f64, _>::zeros((1, 1, 1));
+        let tensor = Tensor::from(arr);
+        let expected = Tensor::new(&[1, 1, 1]).with_values(&[0.0]).unwrap();
+        assert_eq!(expected, tensor);
+
+        let arr = Array::<f32, _>::zeros((1, 2));
+        let tensor = Tensor::from(arr);
+        let expected = Tensor::new(&[1, 2]).with_values(&[0.0, 0.0]).unwrap();
+        assert_eq!(expected, tensor);
+
+        let arr = Array::<i32, _>::zeros((2, 2));
+        let tensor = Tensor::from(arr);
+        let expected = Tensor::new(&[2, 2]).with_values(&[0, 0, 0, 0]).unwrap();
+        assert_eq!(expected, tensor);
+
+        let arr = Array::<u8, _>::zeros((3, 2));
+        let tensor = Tensor::from(arr);
+        let expected = Tensor::new(&[3, 2])
+            .with_values(&[0, 0, 0, 0, 0, 0])
+            .unwrap();
+        assert_eq!(expected, tensor);
+    }
```

Comment:
This could use a test with non-default values so we can verify that the tensor values are actually being set.